packages/react-storage/src/components/StorageBrowser/actions/handlers/copy.ts

```diff
@@ -29,11 +27,11 @@ export const copyHandler: CopyHandler = (input) => {
     credentials,
     customEndpoint,
   } = config;
-  const { key: sourcePath } = data;
+  const { key: sourcePath, fileKey } = data;
 
   const bucket = constructBucket(config);
 
-  const destinationPath = `${path}${getFilenameWithoutPrefix(sourcePath)}`;
+  const destinationPath = `${path}${fileKey}`;
```

Comment:
Updated the approach here to allow to improve consumer ergonomics by providing just the parsed `fileKey` as well as the full `key`